packages/cursorless-engine/src/processTargets/modifiers/PreferredScopeStage.ts

```diff
@@ -0,0 +1,103 @@
+import { type Position, type PreferredScopeModifier } from "@cursorless/common";
+import type { Target } from "../../typings/target.types";
+import type { ModifierStageFactory } from "../ModifierStageFactory";
+import type { ModifierStage } from "../PipelineStages.types";
+import { getContainingScopeTarget } from "./getContainingScopeTarget";
+import type { TargetScope } from "./scopeHandlers/scope.types";
+import type { ScopeHandler } from "./scopeHandlers/scopeHandler.types";
+import type { ScopeHandlerFactory } from "./scopeHandlers/ScopeHandlerFactory";
+
+/**
+ * This modifier tries to find the preferred scope for the target.
+ * It does this by first looking for a containing scope, and if that fails,
+ * it looks for the closest scope.
+ */
+export class PreferredScopeStage implements ModifierStage {
+  constructor(
+    private modifierStageFactory: ModifierStageFactory,
+    private scopeHandlerFactory: ScopeHandlerFactory,
+    private modifier: PreferredScopeModifier,
+  ) {}
+
+  run(target: Target): Target[] {
+    const { scopeType } = this.modifier;
+
+    const scopeHandler = this.scopeHandlerFactory.create(
+      this.modifier.scopeType,
+      target.editor.document.languageId,
+    );
+
+    if (scopeHandler == null) {
+      throw Error(`Couldn't create scope handler for: ${scopeType.type}`);
+    }
+
+    const containingTargets = getContainingScopeTarget(target, scopeHandler);
+
+    if (containingTargets != null) {
+      return containingTargets;
+    }
```

Comment:
```suggestion
    const containingTargets = getContainingScopeTarget(target, scopeHandler);
    if (containingTargets) {
      return containingTargets;
    }
```

Reply:
Since the if statement it is tightly bound to the variable declaration I would remove the blank line for readability

Or if there is an equivalent to the Python syntax:

```
    if (containingTargets := getContainingScopeTarget(target, scopeHandler)) {
      return containingTargets;
    }
```